app/client/src/store/actions.ts

```diff
@@ -61,4 +61,40 @@ export default {
       commit('setStatus', { task: 'network', data: 'submitted' });
     });
   },
+  async getStatus({ commit, state }: { commit: Commit, state: RootState }) {
+    await axios.post(
+      `${config.server_url}/status`,
+      { hash: state.projectHash },
+      {
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      },
+    ).then((response) => {
+      const prev = { ...state.analysisStatus };
+      if (response.data.data.microreact !== prev.microreact) {
+        commit('setStatus', { task: 'microreact', data: response.data.data.microreact });
+      }
+      if (response.data.data.network !== prev.network) {
+        commit('setStatus', { task: 'network', data: response.data.data.network });
+      }
+      if (response.data.data.assign !== prev.assign) {
+        commit('setStatus', { task: 'assign', data: response.data.data.assign });
+      }
```

Comment:
Would there be any downside to just committing the whole thing every time? It would simplify the code if `setStatus` could just take the whole `response.data.data` object and assign it to `state.analysisStatus`

Reply:
I just implemented it like this because I was using the watchers. They would not recognise which values were changed when I overwrite all statuses every time and fire even if a status is the same as before. have changed that now since I removed the watchers.